src/components/Header.tsx

```diff
@@ -0,0 +1,53 @@
+import { useState, useContext } from 'react';
+import classNames from 'classnames';
+import { DispatchContext } from '../TodosContext';
+
+export const Header: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [isHasError, setIsHasError] = useState(false);
+  const onSubmit = useContext(DispatchContext);
+  const handleChandge = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setIsHasError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    setIsHasError(!title);
+
+    if (title) {
+      onSubmit({
+        type: 'add',
+        payload: {
+          id: +new Date(),
+          title,
+          completed: false,
+        },
+      });
+      setTitle('');
+    }
+  };
+
+  return (
+    <header className="header">
+      <h1>todos</h1>
+
+      <form onSubmit={handleSubmit}>
+        <input
+          // eslint-disable-next-line jsx-a11y/no-autofocus
+          autoFocus
+          type="text"
+          data-cy="createTodo"
+          className={classNames(isHasError
+            ? 'new-todo error'
+            : 'new-todo')}
+          placeholder={classNames(isHasError
+            ? 'Please add a text'
+            : 'What needs to be done?')}
```

Comment:
```suggestion
          placeholder={isHasError
            ? 'Please add a text'
            : 'What needs to be done?'}
```
it means you can do it w/o `classNames` function to display the correct placeholder

The same in `className` attribute, but there it looks at least a bit more expected and natural. But usually for such simple cases with ternary, there are not a lot of reasons for using `classnames`. It's handy when you need to add some class if a condition is `true` and do not add anything when a condition is `false`, etc. And of course, in more complicated cases: [examples](https://www.npmjs.com/package/classnames#usage)